packages/xen-api/src/index.js

```diff
@@ -792,7 +812,15 @@ export class Xapi extends EventEmitter {
     }
   }
 
-  async _getHostAddress({ address }) {
+  async _getHostAddress({ address, PIFs, $pool }) {
+    const poolMigrationNetwork = $pool.other_config['xo:migrationNetwork']
+    if (poolMigrationNetwork !== undefined) {
+      const migrationNetworkPifRef = (await this.getRecordByUuid('network', poolMigrationNetwork)).PIFs.filter(pifRef =>
+        PIFs.includes(pifRef)
+      )
+      address = (await this.getRecord('PIF', migrationNetworkPifRef[0])).IP
```

Comment:
Simpler and more efficient:

```suggestion
      address = await this.getField('PIF', migrationNetworkPifRef[0]), 'IP')
```